src/main/scala/strawman/collection/immutable/LazyList.scala

```diff
@@ -36,31 +24,214 @@ class LazyList[+A](expr: => LazyList.Evaluated[A])
   protected[this] def newSpecificBuilder(): Builder[A, LazyList[A]] =
     IndexedSeq.newBuilder().mapResult(_.to(LazyList))
 
-// The following is commented because we need to compare it with the default implementation first
-//  def zipWithIndex: LazyList[(A, Int)] =
-//    LazyList.unfold((0, this)) { case (i, as) =>
-//      as.force.map { case (a, _as) => ((a, i), (i + 1, _as)) }
-//    }
+  /** The stream resulting from the concatenation of this stream with the argument stream.
+    *
+    * @param suffix The collection that gets appended to this lazy list
+    * @return The lazy list containing elements of this lazy list and the iterable object.
+    */
+  def lazyAppendAll[B >: A](suffix: => collection.IterableOnce[B]): LazyList[B] =
+    if (isEmpty) LazyList.fromIterator(suffix.iterator()) else LazyList.cons(head, tail.lazyAppendAll(suffix))
 
   override def className = "LazyList"
 
-  override def toString =
-    if (evaluated)
-      result match {
-        case None => "Empty"
-        case Some((hd, tl)) => s"$hd #:: $tl"
+  override def equals(that: Any): Boolean =
+    if (this eq that.asInstanceOf[AnyRef]) true else super.equals(that)
+
+  override def sameElements[B >: A](that: IterableOnce[B]): Boolean = {
+    @tailrec def lazyListEq(a: LazyList[_], b: LazyList[_]): Boolean =
+      if (a eq b) true else {
+        (a.force, b.force) match {
+          case (Some((ah, at)), Some((bh, bt))) => (ah == bh) && lazyListEq(at, bt)
+          case (None, None) => true
+          case _ => false
+        }
       }
-    else "LazyList(?)"
+    that match {
+      case that: LazyList[_] => lazyListEq(this, that)
+      case _ => super.sameElements(that)
+    }
+  }
+
+  /** Apply the given function `f` to each element of this linear sequence
```

Comment:
This is too much functionality for me to review by eye and have much confidence I've caught a reasonable fraction of any errors.  Hopefully I'll get collections-laws running soon.